### PR TITLE
Unset the LOG_LEVEL variable

### LINKED ISSUE #5217: Super-linter LOG_LEVEL variable conflicts with Checkov and Renovate LOG_LEVEL
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Checkov uses an environment variable named `LOG_LEVEL` to configure the log level, and this conflicts with our `LOG_LEVEL` variable

### Expected Behavior

Don't export the `LOG_LEVEL` variable when running Checkov.

### Super-Linter version

```markdown
v6.0.0
```

### Relevant log output

```shell
2024-02-01T15:35:52.7838131Z [0m2024-02-01 15:35:51 [INFO][0m   Command output for CHECKOV:
2024-02-01T15:35:52.7838705Z ------
2024-02-01T15:35:52.7839016Z Traceback (most recent call last):
2024-02-01T15:35:52.7839549Z   File \"/venvs/checkov/bin/checkov\", line 2, in <module>
2024-02-01T15:35:52.7840122Z     from checkov.main import Checkov
2024-02-01T15:35:52.7841611Z   File \"/venvs/checkov/lib/python3.12/site-packages/checkov/main.py\", line 21, in <module>
2024-02-01T15:35:52.7842774Z     import checkov.logging_init  # noqa  # should be imported before the others to ensure correct logging setup
2024-02-01T15:35:52.7843760Z     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-02-01T15:35:52.7844765Z   File \"/venvs/checkov/lib/python3.12/site-packages/checkov/logging_init.py\", line 8, in <module>
2024-02-01T15:35:52.7845554Z     logging.basicConfig(level=LOG_LEVEL)
2024-02-01T15:35:52.7846246Z   File \"/usr/local/lib/python3.12/logging/__init__.py\", line 2144, in basicConfig
2024-02-01T15:35:52.7846923Z     root.setLevel(level)
2024-02-01T15:35:52.7847496Z   File \"/usr/local/lib/python3.12/logging/__init__.py\", line 1514, in setLevel
2024-02-01T15:35:52.7848170Z     self.level = _checkLevel(level)
2024-02-01T15:35:52.7848579Z                  ^^^^^^^^^^^^^^^^^^
2024-02-01T15:35:52.7849217Z   File \"/usr/local/lib/python3.12/logging/__init__.py\", line 213, in _checkLevel
2024-02-01T15:35:52.7850308Z     raise ValueError(\"Unknown level: %r\" % level)
2024-02-01T15:35:52.7851095Z ValueError: Unknown level: 'TRACE'
2024-02-01T15:35:52.7851550Z ------[0m
2024-02-01T15:35:52.7852181Z [0m2024-02-01 15:35:51 [DEBUG][0m   Ending GitHub Actions log group: CHECKOV[0m
```

### Steps To Reproduce

1. Set `LOG_LEVEL` to `TRACE`
2. Run super-linter

### Anything else?

#5215

### PR DESCRIPTION
**Pull request by ferrarimarco**
# Proposed changes

- Super-linter uses the LOG_LEVEL variable to let the user configure the desired log level. Checkov and Renovate use a variable with the same name for the same purpose, but accept a different set of values, and exit with an error if it gets an unknown value for that variable.
- Refactor the VERBOSE log level to the more commonly used INFO. Configuration validation will warn users if they use VERBOSE and instruct them to use INFO instead. This is not a breaking change because super-linter falls back on INFO if VERBOSE is set.
- Remove the TRACE log level because we rarely used it. As with VERBOSE, configuration validation will warn the user. Fall back to DEBUG if the user configured LOG_LEVEL to VERBOSE.

Close #5217

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I wrote the necessary upgrade instructions in the [upgrade guide](../docs/upgrade-guide.md).
- [x] If this pull request is about and existing issue,
  I added the `Fix #ISSUE_NUMBER` label to the description of the pull request.

### Super-linter maintainer tasks

- [x] Label as `breaking` if this change breaks compatibility with the previous released version.
- [x] Label as either: `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`.

### CODE REVIEW

**Comment by ferrarimarco** [comment]
(pushing a test that's expected to fail. I'll push the fix after that)

File: lib/functions/validation.sh [hunk 1]
```diff
@@ -259,15 +259,37 @@ function WarnIfVariableIsSet() {
   local INPUT_VARIABLE_NAME="${1}"
 
   if [ -n "${INPUT_VARIABLE:-}" ]; then
-    warn "${INPUT_VARIABLE_NAME} environment variable is set, it's deprecated, and super-linter will ignore it. Remove it from your configuration. This warning may turn in a fatal error in the future."
+    warn "${INPUT_VARIABLE_NAME} environment variable is set, it's deprecated, and super-linter will ignore it. Remove it from your configuration. This warning may turn in a fatal error in the future. For more information, see the upgrade guide: https://github.com/super-linter/super-linter/blob/main/docs/upgrade-guide.md"
```
Comment (Hanse00) [inline lib/functions/validation.sh:262]: We should probably add an issue to actually add more info in the upgrade guide.

**Review by Hanse00 — APPROVED** [review]
Looks good to me, though I'm curious if @zkoppert has any opinions on this.

File: lib/functions/validation.sh (hunk 1 above)
Comment (ferrarimarco) [reply lib/functions/validation.sh:262]: SGTM. Can you please open one listing the stuff we want to put there? Thanks!

**Pull request merged**